android/src/main/java/com/tanguyantoine/react/MusicControlModule.java

```diff
@@ -212,33 +203,47 @@ synchronized public void setNowPlaying(ReadableMap metadata) {
         nb.setContentInfo(album);
         nb.setColor(notificationColor);
 
-        if(artwork != null) {
-            final String artworkUrl = artwork;
-            final boolean artworkLocal = localArtwork;
-
-            artworkThread = new Thread(new Runnable() {
-                @Override
-                public void run() {
-                    Bitmap bitmap = loadArtwork(artworkUrl, artworkLocal);
-                    md.putBitmap(MediaMetadataCompat.METADATA_KEY_ART, bitmap);
-                    nb.setLargeIcon(bitmap);
-
-                    session.setMetadata(md.build());
-                    notification.show(nb, isPlaying);
-                    artworkThread = null;
-                }
-            });
-            artworkThread.start();
+        if (metadata.hasKey("artwork")) {
+          setArtwork(metadata);
         } else {
-            md.putBitmap(MediaMetadataCompat.METADATA_KEY_ART, null);
-            nb.setLargeIcon(null);
+          md.putBitmap(MediaMetadataCompat.METADATA_KEY_ART, null);
+          nb.setLargeIcon(null);
```

Comment:
Android Java binaries work differently than regular java binaries, the methods are stored in a different way, the problem is that they have a limit of methods. It is possible to workaround it, but it's kinda annoying, mainly for someone that is using React Native and have no idea about how Android works.